src/MELT/MELT.csproj

```diff
@@ -17,5 +17,6 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.1.0" />
     <PackageReference Include="Microsoft.Extensions.Logging" Version="2.1.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.5.0" />
```

Comment:
Is bringing the extra dependency of Immutable really worth it? would be nice to keep dependencies limited unless it gives you a great benefit. Like I would have probably used a ConcurrentQueue instead, but not sure yet if even that is worth it.

Reply:
Ok. I've changed it to a list (I had to use a lock anyway so there was no reason to use a ConcurrentQueue).